### PR TITLE
Adding Events method

### LINKED ISSUE #101: Events are not an implemented Class.  Generic get method
Implement an events class

Add a get(params) method so that we can pass in our own params instead of the get_*'s that apply the data to only the filters. 

params = {
  :hostids => '1'
}
get(params)

### PR DESCRIPTION
**Pull request by kevin-j-smith**
Signed-off-by: Kevin J. Smith <kevin.j.smith2@cerner.com>

closes #101 Closes #101

### CODE REVIEW

**Comment by Splyth** [comment]
The code is A-OK. Could you add some spec tests for it. We're trying to enforce it going forward. while we clean up and get coverage on the code that already exists.

Once done. I'll happily +1 it.

**Comment by kevin-j-smith** [comment]
Yes, I can do that as well as add method comments for you docs.   Thanks for the reply.

**Comment by kevin-j-smith** [comment]
@Splyth,  In review of the PR I have supplied a spec/event.rb file to cover some basic testing.  This was mainly copied from other spec tests.

**Comment by Splyth** [comment]
Looks like travis has problem with your gemfile:
Bundler could not find compatible versions for gem "ruby":

**Comment by kevin-j-smith** [comment]
Looks like the travis file is still using unsupported versions of ruby as required-to-pass validation 'locks'.   Updating to be more modern.

**Comment by kevin-j-smith** [comment]
@Splyth @snehitgajjar, All of the tests have been updated and the API only tweaked to match Zabbix 4.0, which the tests are using.  Hopefully, this sets a good foot forward as tests have been failing for a significant amount of time and PRs have still been merged.

File: lib/zabbixapi/basic/basic_alias.rb [hunk 1]
```diff
@@ -7,6 +7,7 @@ class Basic
     # @raise [HttpError] Error raised when HTTP status from Zabbix Server response is not a 200 OK.
     # @return [Hash]
     def get(data)
+      puts "HERE"
```
Comment (Splyth) [inline lib/zabbixapi/basic/basic_alias.rb:10]: is this puts statement needed?

File: lib/zabbixapi/basic/basic_logic.rb [hunk 2]
```diff
@@ -89,6 +89,16 @@ def get_full_data(data)
...
+    # DUPLICATE METHOD
+    # def get(data)
+    #   log "[DEBUG] Call get with parameters: #{data.inspect}"
+
+    #   @client.api_request(
+    #     method: "#{method_name}.get",
+    #     params: data
+    #   )
+    # end
+
```
Comment (Splyth) [inline lib/zabbixapi/basic/basic_logic.rb:101]: Is this still needed?

File: lib/zabbixapi/basic/basic_alias.rb (hunk 1 above)
Comment (kevin-j-smith) [reply lib/zabbixapi/basic/basic_alias.rb:10]: Nope sorry, testing output.

File: lib/zabbixapi/basic/basic_logic.rb (hunk 2 above)
Comment (kevin-j-smith) [reply lib/zabbixapi/basic/basic_logic.rb]: Nope, the get method was abstracted in the alias class.

**Pull request merged**